src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,113 +29,214 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
+     * the unit of right with the list of left
+     * or the unit of left with the list of Right
+     */
     @Override
-    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u){
-        if (!(isSinkStatement(u) || isSourceStatement(u))){
+    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u) {
+        if (!(isRightStatement(u) || isLeftStatement(u))) {
             return;
         }
+
         List<DataFlowAbstraction> left = new ArrayList<>();
         List<DataFlowAbstraction> right = new ArrayList<>();
 
-        u.getUseAndDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-            String localName = getLocalName((Local) v.getValue());
-            for (DataFlowAbstraction filterIn: in){
-                String inName = getLocalName(filterIn.getLocal());
+        getStatementAssociatedWithUnit(u).forEach(valueBox -> {
+            String localName = getLocalValueBoxName(valueBox);
+
+            for (DataFlowAbstraction filterIn : in) {
+                String inName = getLocalDataFlowAbstraction(filterIn);
 
-                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)){
+                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)) {
                     left.add(filterIn);
-                }else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)){
+                } else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)) {
                     right.add(filterIn);
                 }
             }
         });
 
-        if(isSinkStatement(u)) {
+        if (isRightStatement(u)) {
             checkConflicts(u, left);
-        }else  if (isSourceStatement(u)){
+        } else if (isLeftStatement(u)) {
             checkConflicts(u, right);
         }
     }
 
-    protected void checkConflicts(Unit u, List<DataFlowAbstraction> statements){
-        for (DataFlowAbstraction statement : statements) {
-            if (statementEquals(statement, u)) {
-                Conflict c = new Conflict(statement.getStmt(), findStatement(u));
+    /*
+     * Checks if there is a conflict and if so adds it to the conflict list.
+     */
+    private void checkConflicts(Unit u, List<DataFlowAbstraction> dataFlowAbstractionList) {
+        for (DataFlowAbstraction dataFlowAbstraction : dataFlowAbstractionList) {
+            if (checkEqualVariablesName(dataFlowAbstraction, u)) {
+                Conflict c = new Conflict(dataFlowAbstraction.getStmt(), findStatement(u));
                 Collector.instance().addConflict(c);
             }
         }
     }
+    /*
+     * Checks the equality of variable names between an
+     * item in the abstraction list and all the UnitBox's ValueBox.
+     */
+    private boolean checkEqualVariablesName(DataFlowAbstraction dataFlowAbstraction, Unit u) {
```

Comment:
@barbosamaatheus this still is not a good name

would abstractionVariableIsInIUnitDefBoxes be better?